tools/slims.py

```diff
@@ -0,0 +1,245 @@
+import os
+import json
+from datetime import datetime
+
+from slims.slims import Slims
+from slims.criteria import is_one_of, equals, conjunction
+from slims.content import Status
+
+import passwords
+
+# Initial set up of the SLIMS instance
+instance = 'wopr_query'
+url = passwords.slims.url
+user = passwords.slims.user
+password = passwords.slims.password
+slims = Slims(instance, url, user, password)
+
+
+class SlimsSample:
+    """
+    Class containing the sample data.
+    """
+    def __init__(self, sample_name, run_tag=''):
+        self.sample_name = sample_name
+        self.run_tag = run_tag
+
+        self._dna = None
+        self._fastq = None
+        self._fastqs = []
+        self._bioinformatics = None
+
+    @property
+    def dna(self):
+        if not self._dna:
+            records = slims.fetch('Content', conjunction()
+                                  .add(equals('cntn_id', self.sample_name))
+                                  .add(equals('cntn_fk_contentType', 6)))
+
+            if len(records) > 1:
+                raise Exception('More than 1 DNA somehow.')
+
+            if records:
+                self._dna = records[0]
+
+        return self._dna
+
+    @property
+    def fastq(self):
+        if not self.run_tag:
+            raise Exception('Can not fetch fastq without a set run tag.')
+
+        if not self._fastq:
+            records = slims.fetch('Content', conjunction()
+                                  .add(equals('cntn_id', self.sample_name))
+                                  .add(equals('cntn_fk_contentType', 22))
+                                  .add(equals('cntn_cstm_runTag', self.run_tag)))
+
+            if len(records) > 1:
+                raise Exception('More than 1 fastq somehow.')
+
+            if records:
+                self._fastq = records[0]
+
+        return self._fastq
+
+    @property
+    def fastqs(self):
```

Comment:
is this for samples that have fastqs in more than one run? can you have more than one pair of fastq files as input to the pipeline for a sample?

Reply:
Yupp, exactly. The wrapper takes all fastq objects belonging to the sample.